### PR TITLE
create DB based on diagram

### PR DESCRIPTION
**Pull request by Booze33**
<h1>Hello!</h1>

<h2>In Today's Project...</h2>

- We created a database 
- We used the diagram to create the tables

### CODE REVIEW

File: schema_based_on_diagram.sql [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+CREATE TABLE invoices (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP NOT NULL,
+  payed_at TIMESTAMP NOT NULL,
+  medical_history_id INT REFERENCES medical_histories(id)
+);
+
+CREATE TABLE treatments (
+  id INT REFERENCES medical_histories(id) ON DELETE CASCADE,
```
Comment (brytebee) [inline schema_based_on_diagram.sql:24]: This should be the id of the `treatment` table.

`medical_history` and `treatment` tables share a many-to-many relationship and so it is required that a `join table` is created for them though not shown in the ERD. You could name this table `medical_history_has_treatment`.

The table would have foreign keys to reference both tables. Also indexes would be created for them.

File: schema_based_on_diagram.sql
Comment (brytebee) [inline schema_based_on_diagram.sql:1]: For all foreign keys added, you are required to create indexes.

image

**Review by brytebee — CHANGES_REQUESTED** [review]
Hi, @Booze33 & @christianonoh👋,

##

### Highlights💡
- Good commit message.✔️ 
- Good PR Summary.✔️
- Great DB design.✔️
- Great practice of model set-up and relationships.✔️
- App functions as it should.✔️

##

Good job so far coming so far with the progress you've made! 💪💪
 
However, there are some issues that you still need to work on to go to the next project but you are almost there!



# Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Please, remember to tag me in your question so I can receive the notification.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @brytebee in your question so I can receive the notification.
### _**Use @brytebee on slack for faster responses.**_


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  quantity INT, total_price DECIMAL NOT NULL,
+  invoice_id INT REFERENCES invoices(id) ON DELETE CASCADE,
+  treatment_id INT REFERENCES treatments(id) ON DELETE CASCADE
+);
+
+CREATE TABLE medical_histories_has_treatments (
+  medical_histories_id INT REFERENCES medical_histories(id),
+  treatments_id INT REFERENCES treatments(id),
+  PRIMARY KEY(medical_histories_id, treatments_id)
+);
```
Comment (osamaashraf6) [inline schema_based_on_diagram.sql]: - [ ] While you make a great effort 😃, as stated by the previous reviewer I see that you forgot to make all FOREIGN KEYS have indexes created, as you know the importance of indexing for the optimization of the SQL queries, so please 💯 try to add the index for all the foreign keys to stick to the requirements and also for enhancing the performance 🚀💯

image


- Hint: 

```snippet
CREATE INDEX ON medical_histories (patient_id);
CREATE INDEX ON invoices (medical_history_id);
CREATE INDEX ON invoice_items (invoice_id);
CREATE INDEX ON invoice_items (treatment_id);
CREATE INDEX ON medical_histories_has_treatments (medical_history_id);
CREATE INDEX ON medical_histories_has_treatments (treatment_id);
```

**Review by osamaashraf6 — CHANGES_REQUESTED** [review]
Hi @Booze33  😊, 

Good job so far!
There are some issues that you still need to work on from he previous reviewer to go to the next project but you are almost there!
SoCloseAlmostGIF (3)

## Highlights🎉🎉

- [ ] Good using the SQL commands 🚀
- [ ] Good converting the schema into the SQL code👌
- [ ] Good adding the relation many to many.💯

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
- N/A
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: schema_based_on_diagram.sql [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+  patient_id INT REFERENCES patients(id) ON DELETE CASCADE,
+  status VARCHAR(255) NOT NULL
+);
+
+CREATE TABLE invoices (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  total_amount DECIMAL,
+  generated_at TIMESTAMP NOT NULL,
+  payed_at TIMESTAMP NOT NULL,
+  medical_history_id INT REFERENCES medical_histories(id)
```
Comment (descholar-ceo) [inline schema_based_on_diagram.sql:20]: -  [ ] This `medical_history_id` also needs an index on it

File: schema_based_on_diagram.sql [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  id INT GENERATED ALWAYS AS IDENTITY,
+  name VARCHAR(255) NOT NULL,
+  date_of_birth DATE,
+  PRIMARY KEY (id)
+);
+
+CREATE TABLE medical_histories (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  admitted_at TIMESTAMP,
+  patient_id INT REFERENCES patients(id) ON DELETE CASCADE,
```
Comment (descholar-ceo) [inline schema_based_on_diagram.sql:11]: - [ ] This `patient_id` needs an index

File: schema_based_on_diagram.sql [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  type VARCHAR(255) NOT NULL,
+  name VARCHAR(255) NOT NULL
+)
+
+CREATE TABLE invoice_item (
+  id INT GENERATED ALWAYS AS IDENTITY,
+  unit_price DECIMAL NOT NULL,
+  quantity INT, total_price DECIMAL NOT NULL,
+  invoice_id INT REFERENCES invoices(id) ON DELETE CASCADE,
+  treatment_id INT REFERENCES treatments(id) ON DELETE CASCADE
```
Comment (descholar-ceo) [inline schema_based_on_diagram.sql:35]: -  [ ] These `invoice_id`, and `treatment_id` need indexes on them

**Review by descholar-ceo — CHANGES_REQUESTED** [review]
Hi @Booze33 && @christianonoh 🤝 🤝 , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights
👍🏽 Correct git flow usage ✔️ 
👍🏽 Readme file follows microverse's template ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by uwadonat — APPROVED** [review]
## Hi @Booze33,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- Correct workflow was used ✔️
- Well organized and clean code ✔️
approv

# Status: Approved ✔️✔️✔️

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**